Haml Setup

Petercopter opened the pull request:
Partial fix for #45 

Will require changes in Syntax Tree as well

kddnewton commented:
Nice! This looks right. Let's get the other piece of this merged into syntax tree first and then we can merge this.

Petercopter commented:
@kddnewton This is what I've got so far. 

I added the `onLanguage` event like you mentioned, then I thrashed around a lot and added the `documentSelector` as well. That seems to get me to the point where VSCode is using Syntax Tree to trigger.

This may or may not be doing something in the config:
```
"[haml]": {
    "editor.defaultFormatter": "ruby-syntax-tree.vscode-syntax-tree"
  }
```
At this point I was going to switch over to Syntax Tree and try to figure out how to get Haml formatting working there, does that sound right?

Petercopter commented:
https://github.com/ruby-syntax-tree/syntax_tree/pull/127 is merged :D
